openwhisk/executor.go

```diff
@@ -108,15 +135,54 @@ func (proc *Executor) Start() error {
 		return fmt.Errorf("command exited")
 	}
 	Debug("pid: %d", proc.cmd.Process.Pid)
+
 	go func() {
 		proc.cmd.Wait()
 		proc.exited <- true
 	}()
+
+	// not waiting for an ack, so use a timeout
+	if !waitForAck {
+		select {
+		case <-proc.exited:
+			return fmt.Errorf("command exited")
+		case <-time.After(DefaultTimeoutStart):
+			return nil
+		}
+	}
+
+	// wait for acknowledgement
+	Debug("waiting for an ack")
+	ack := make(chan error)
```

Comment:
this isn't guarded by an else on `if !waitForAck` - how come?

Reply:
No it is not guarded. If there is not the OW_WAIT_FOR_ACK variable, the proxy behaves as before so it waits for a timeout OR a termination then returns. Otherwise it waits for an ack.